Sources/ConcurrencyHelpers/ConcurrentMap.swift

```diff
@@ -0,0 +1,84 @@
+extension RandomAccessCollection where Self: Sendable, Element: Sendable {
+    func concurrentMap<B: Sendable>(minBatchSize: Int = 4096, _ transform: @Sendable @escaping (Element) async -> B) async -> [B] {
+        precondition(minBatchSize >= 1)
+        let n = self.count
+        let batchCount = (n + minBatchSize - 1) / minBatchSize
+        if batchCount < 2 {
+            return await withTaskGroup(of: B.self) { group in
+                var results: [B] = []
+                for element in self {
+                    group.addTask { await transform(element) }
+                }
+                for await result in group {
```

Comment:
This looks incorrect to me.
See https://developer.apple.com/documentation/swift/taskgroup/next()

>The values returned by successive calls to this method appear in the order that the tasks completed, not in the order that those tasks were added to the task group.